controllers/ca_controller.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/kubewarden/kubewarden-controller/internal/pkg/admission"
+	"github.com/kubewarden/kubewarden-controller/internal/pkg/admissionregistration"
+	"github.com/kubewarden/kubewarden-controller/internal/pkg/constants"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// This controller reconcile the root ca used by the Kubewarden stack
+type KubewardenCAReconciler struct {
+	client.Client
+	Log        logr.Logger
+	Scheme     *runtime.Scheme
+	Reconciler admission.Reconciler
+}
+
+func (r *KubewardenCAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	CASecret, initialized, err := r.Reconciler.FetchOrInitializeRootCASecret(ctx, admissionregistration.GenerateCA, admissionregistration.PemEncodeCertificate)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "failed to create root CA")
+	}
+	// if we have a new kubewarden root ca it is necessary to update the policy server secrets.
+	if initialized {
+		r.Log.Info("Root CA initialized")
+		err := r.Reconciler.UpdateAllPolicyServerSecrets(ctx, CASecret)
```

Comment:
I don't know if this method is doing that, but this is a list of steps that have to be done when the CA is generated/changed:

* Regenerate all the certificates used by the policy servers
* Update the contents of the Secret used by the policy server: add the new certificate
* Rollout the policy servers to ensure they are restarted using the new certificates
* Update all the validating/mutating webhook configurations generated by KW: there's the old CA bundled in there, they have to consume the new one

Reply:
This method is regenerating all the policy server secrets. However, it's not necessary to rollout the policy servers. Acordially to the Kubernetes [docs](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod) the volume will be update with the latest content of the secret:

> When a volume contains data from a Secret, and that Secret is updated, Kubernetes tracks this and updates the data in the volume, using an eventually-consistent approach

I'll double check the webhooks

Reply:
Ah, but the policy server just read the secret content in the startup. Yeah, a rollout will be necessary unless we find a way to signal policy server to reread the secret. Which is out of scope of this PR for now. I'll ensure that server will be restarted.

Reply:
> I'll double check the webhooks

Yesterday I was in a hurry while doing the review. You have to change the contents of [caBundle](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#webhookclientconfig-v1-admissionregistration-k8s-io) to be the new CA